compilation problem in the integration test matMul

I have this compilation problem with the matMul example:

####################################################################

[ 80%] Building NVCC (Device) object matMul/CMakeFiles/matMul.dir/src/matMul_generated_main.cpp.o
/home/ivan/Desktop/Alpaka_git/alpaka/test/integ/matMul/src//main.cpp(297) (col. 15): internal error: assertion failed at: "/dvs/p4/build/sw/rel/gpu_drv/r361/r361_00/drivers/compiler/edg/EDG_4.10/src/il_to_str.c", line 7191


1 catastrophic error detected in the compilation of "/tmp/tmpxft_00003f69_00000000-5_main.cpp4.ii".
Compilation aborted.
Aborted (core dumped)
CMake Error at matMul_generated_main.cpp.o.cmake:282 (message):
  Error generating file
  /home/ivan/Desktop/Alpaka_git/alpaka/test/integ/build/matMul/CMakeFiles/matMul.dir/src/./matMul_generated_main.cpp.o


matMul/CMakeFiles/matMul.dir/build.make:63: recipe for target 'matMul/CMakeFiles/matMul.dir/src/matMul_generated_main.cpp.o' failed
make[2]: *** [matMul/CMakeFiles/matMul.dir/src/matMul_generated_main.cpp.o] Error 1
CMakeFiles/Makefile2:332: recipe for target 'matMul/CMakeFiles/matMul.dir/all' failed
make[1]: *** [matMul/CMakeFiles/matMul.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2

#####################################################################

thanks in advance for any help!

Could you please post which OS do you have used and which compiler, boost and CUDA (if used).
And all commands used up to the broken compile.

Am 16. November 2017 09:37:37 MEZ, schrieb ivandrodri <notifications@github.com>:
>I have this compilation problem with the matMul example:
>
>####################################################################
>
>[ 80%] Building NVCC (Device) object
>matMul/CMakeFiles/matMul.dir/src/matMul_generated_main.cpp.o
>/home/ivan/Desktop/Alpaka_git/alpaka/test/integ/matMul/src//main.cpp(297)
>(col. 15): internal error: assertion failed at:
>"/dvs/p4/build/sw/rel/gpu_drv/r361/r361_00/drivers/compiler/edg/EDG_4.10/src/il_to_str.c",
>line 7191
>
>
>1 catastrophic error detected in the compilation of
>"/tmp/tmpxft_00003f69_00000000-5_main.cpp4.ii".
>Compilation aborted.
>Aborted (core dumped)
>CMake Error at matMul_generated_main.cpp.o.cmake:282 (message):
>  Error generating file
>/home/ivan/Desktop/Alpaka_git/alpaka/test/integ/build/matMul/CMakeFiles/matMul.dir/src/./matMul_generated_main.cpp.o
>
>
>matMul/CMakeFiles/matMul.dir/build.make:63: recipe for target
>'matMul/CMakeFiles/matMul.dir/src/matMul_generated_main.cpp.o' failed
>make[2]: ***
>[matMul/CMakeFiles/matMul.dir/src/matMul_generated_main.cpp.o] Error 1
>CMakeFiles/Makefile2:332: recipe for target
>'matMul/CMakeFiles/matMul.dir/all' failed
>make[1]: *** [matMul/CMakeFiles/matMul.dir/all] Error 2
>Makefile:83: recipe for target 'all' failed
>make: *** [all] Error 2
>
>#####################################################################
>
>thanks in advance for any help!
>
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/ComputationalRadiationPhysics/alpaka/issues/446

-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.

I am using ubuntu 16.04, boost 1.65 and CUDA 8.0 .   I went to matMul  and create a build folder to generate the project with "cmake .." . Cmake works but when I run make this is the full output:


ivan@ivan-XPS-15-9560:~/Desktop/Alpaka/alpaka/test/integ/matMul/build$ make
[ 33%] Built target alpaka
[ 66%] Built target common
[ 83%] Building NVCC (Device) object CMakeFiles/matMul.dir/src/matMul_generated_main.cpp.o
/home/ivan/Desktop/Alpaka_git/alpaka/test/integ/matMul/src//main.cpp(299): error: argument list for class template "alpaka::mem::view::ViewPlainPtr" is missing
          detected during:
            instantiation of "auto alpaka::meta::detail::ForEachTypeHelper<TList<T, Ts...>>::forEachTypeHelper(TFnObj &&, TArgs &&...)->void [with TList=std::tuple, T=alpaka::test::acc::detail::AccCpuSerialIfAvailableElseVoid<alpaka::dim::DimInt<2UL>, uint32_t>, Ts=<alpaka::test::acc::detail::AccCpuThreadsIfAvailableElseVoid<alpaka::dim::DimInt<2UL>, uint32_t>, alpaka::test::acc::detail::AccCpuOmp2BlocksIfAvailableElseVoid<alpaka::dim::DimInt<2UL>, uint32_t>, alpaka::test::acc::detail::AccCpuOmp2ThreadsIfAvailableElseVoid<alpaka::dim::DimInt<2UL>, uint32_t>, alpaka::test::acc::detail::AccGpuCudaRtIfAvailableElseVoid<alpaka::dim::DimInt<2UL>, uint32_t>>, TFnObj=MatMulTester &, TArgs=<uint32_t &, uint32_t &, uint32_t &>]" 
/home/ivan/Desktop/Alpaka_git/alpaka/include/alpaka/meta/ForEachType.hpp(122): here
            instantiation of "auto alpaka::meta::forEachType<TList,TFnObj,TArgs...>(TFnObj &&, TArgs &&...)->void [with TList=alpaka::test::acc::EnabledAccs<alpaka::dim::DimInt<2UL>, uint32_t>, TFnObj=MatMulTester &, TArgs=<uint32_t &, uint32_t &, uint32_t &>]" 
(434): here

1 error detected in the compilation of "/tmp/tmpxft_00004d24_00000000-7_main.cpp1.ii".
CMake Error at matMul_generated_main.cpp.o.cmake:282 (message):
  Error generating file
  /home/ivan/Desktop/Alpaka_git/alpaka/test/integ/matMul/build/CMakeFiles/matMul.dir/src/./matMul_generated_main.cpp.o


CMakeFiles/matMul.dir/build.make:910: recipe for target 'CMakeFiles/matMul.dir/src/matMul_generated_main.cpp.o' failed
make[2]: *** [CMakeFiles/matMul.dir/src/matMul_generated_main.cpp.o] Error 1
CMakeFiles/Makefile2:142: recipe for target 'CMakeFiles/matMul.dir/all' failed
make[1]: *** [CMakeFiles/matMul.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2

Do you use the master or dev branch?

Am 16. November 2017 10:39:59 MEZ, schrieb ivandrodri <notifications@github.com>:
>I am using ubuntu 16.04, boost 1.65 and CUDA 8.0 .   I went to matMul 
>and create a build folder to generate the project with "cmake .." .
>Cmake works but when I run make this is the full output:
>
>
>ivan@ivan-XPS-15-9560:~/Desktop/Alpaka/alpaka/test/integ/matMul/build$
>make
>[ 33%] Built target alpaka
>[ 66%] Built target common
>[ 83%] Building NVCC (Device) object
>CMakeFiles/matMul.dir/src/matMul_generated_main.cpp.o
>/home/ivan/Desktop/Alpaka_git/alpaka/test/integ/matMul/src//main.cpp(299):
>error: argument list for class template
>"alpaka::mem::view::ViewPlainPtr" is missing
>          detected during:
>instantiation of "auto alpaka::meta::detail::ForEachTypeHelper<TList<T,
>Ts...>>::forEachTypeHelper(TFnObj &&, TArgs &&...)->void [with
>TList=std::tuple,
>T=alpaka::test::acc::detail::AccCpuSerialIfAvailableElseVoid<alpaka::dim::DimInt<2UL>,
>uint32_t>,
>Ts=<alpaka::test::acc::detail::AccCpuThreadsIfAvailableElseVoid<alpaka::dim::DimInt<2UL>,
>uint32_t>,
>alpaka::test::acc::detail::AccCpuOmp2BlocksIfAvailableElseVoid<alpaka::dim::DimInt<2UL>,
>uint32_t>,
>alpaka::test::acc::detail::AccCpuOmp2ThreadsIfAvailableElseVoid<alpaka::dim::DimInt<2UL>,
>uint32_t>,
>alpaka::test::acc::detail::AccGpuCudaRtIfAvailableElseVoid<alpaka::dim::DimInt<2UL>,
>uint32_t>>, TFnObj=MatMulTester &, TArgs=<uint32_t &, uint32_t &,
>uint32_t &>]" 
>/home/ivan/Desktop/Alpaka_git/alpaka/include/alpaka/meta/ForEachType.hpp(122):
>here
>instantiation of "auto
>alpaka::meta::forEachType<TList,TFnObj,TArgs...>(TFnObj &&, TArgs
>&&...)->void [with
>TList=alpaka::test::acc::EnabledAccs<alpaka::dim::DimInt<2UL>,
>uint32_t>, TFnObj=MatMulTester &, TArgs=<uint32_t &, uint32_t &,
>uint32_t &>]" 
>(434): here
>
>1 error detected in the compilation of
>"/tmp/tmpxft_00004d24_00000000-7_main.cpp1.ii".
>CMake Error at matMul_generated_main.cpp.o.cmake:282 (message):
>  Error generating file
>/home/ivan/Desktop/Alpaka_git/alpaka/test/integ/matMul/build/CMakeFiles/matMul.dir/src/./matMul_generated_main.cpp.o
>
>
>CMakeFiles/matMul.dir/build.make:910: recipe for target
>'CMakeFiles/matMul.dir/src/matMul_generated_main.cpp.o' failed
>make[2]: *** [CMakeFiles/matMul.dir/src/matMul_generated_main.cpp.o]
>Error 1
>CMakeFiles/Makefile2:142: recipe for target 'CMakeFiles/matMul.dir/all'
>failed
>make[1]: *** [CMakeFiles/matMul.dir/all] Error 2
>Makefile:83: recipe for target 'all' failed
>make: *** [all] Error 2
> 
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/ComputationalRadiationPhysics/alpaka/issues/446#issuecomment-344868228

-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.

--- a/script/travis/run_tests.sh
+++ b/script/travis/run_tests.sh
@@ -53,7 +53,7 @@ BOOST_TEST_OPTIONS="--log_level=test_suite --color_output=true"
 ./script/travis/compileExec.sh "test/integ/axpy/" ./axpy
 if [ "${ALPAKA_ACC_GPU_CUDA_ONLY_MODE}" == "ON" ] ;then ./script/travis/compileExec.sh "test/integ/cudaOnly/" ./cudaOnly ;fi
 ./script/travis/compileExec.sh "test/integ/mandelbrot/" ./mandelbrot
-#./script/travis/compileExec.sh "test/integ/matMul/" ./matMul
+./script/travis/compileExec.sh "test/integ/matMul/" ./matMul
 ./script/travis/compileExec.sh "test/integ/sharedMem/" ./sharedMem
 
 #-------------------------------------------------------------------------------

--- a/test/integ/matMul/src/main.cpp
+++ b/test/integ/matMul/src/main.cpp
@@ -35,11 +35,11 @@
 //! A matrix multiplication kernel.
 //! Computes C += A*B. LxM * MxN -> LxN
 //! This is an adaption of the algorithm from the CUDA developers guide.
-//! \tparam TAcc The accelerator environment to be executed on.
 class MatMulKernel
 {
 public:
     //-----------------------------------------------------------------------------
+    //! \tparam TAcc The accelerator environment to be executed on.
     //! \tparam TElem The matrix element type.
     //! \param acc The accelerator to be executed on.
     //! \param m The height of the A matrix.
@@ -232,13 +232,14 @@ struct MatMulTester
         using PltfAcc = alpaka::pltf::Pltf<DevAcc>;
         using StreamAcc = alpaka::test::stream::DefaultStream<alpaka::dev::Dev<TAcc>>;
         using PltfHost = alpaka::pltf::PltfCpu;
+        using DevHost = alpaka::dev::Dev<PltfHost>;
         using StreamHost = alpaka::stream::StreamCpuAsync;
 
         // Create the kernel function object.
         MatMulKernel kernel;
 
         // Get the host device.
-        auto const devHost(
+        DevHost const devHost(
             alpaka::pltf::getDevByIdx<PltfHost>(0u));
 
         // Get a stream on the host device.
@@ -246,7 +247,7 @@ struct MatMulTester
             devHost);
 
         // Select a device to execute on.
-        auto const devAcc(
+        DevAcc const devAcc(
             alpaka::pltf::getDevByIdx<PltfAcc>(0u));
 
         // Get a stream on the accelerator device.
@@ -295,7 +296,7 @@ struct MatMulTester
         // For multi dimensional data you could directly create them using alpaka::mem::buf::alloc<Type>(devHost, extent), which is not used here.
         // Instead we use ViewPlainPtr to wrap the data.
         using BufWrapper = alpaka::mem::view::ViewPlainPtr<
-            std::decay<decltype(devHost)>::type,
+            DevHost,
             Val,
             alpaka::dim::DimInt<2u>,
             TSize>;

